src/content/docs/en/guides/cms/keystatic.mdx

```diff
@@ -2,13 +2,263 @@
+    
+    <Keystatic client:only />
+    ```
+
+3. Create a new file called `src/pages/api/keystatic/[...params].ts` that will create `API routes` for Keystatic's Admin UI.
+
+    ```jsx
+    // src/pages/api/keystatic/[...params].ts
+    import { makeHandler } from '@keystatic/astro/api'
+    import keystaticConfig from '../../../../keystatic.config'
+    
+    export const all = makeHandler({
+      config: keystaticConfig,
+    })
+    
+    export const prerender = false
+    ```
+
+4. To launch your Keystatic Admin UI dashboard, start Astro's dev server:
+
+    ```bash
+    npm run dev
+    ```
+
+5. Visit `http://127.0.0.1:3000/keystatic` in the browser to see the Keystatic Admin UI running.
+
+## Creating a new post
+
+Try and visit the `/keystatic` page in the browser one more time, and click on the “Posts” collection.
+
+Go ahead, and create a new post.
+
+:::note
+In our Keystatic config file, you've set the `path` property for our `posts` collection to `src/content/posts/*`.
+
+As a result, creating a new post from the Keystatic Admin UI should create a new `content` directory in the `src` directory, with the new post `.mdoc` file inside.
+:::
+
+If everything worked correctly, you'll find your new post inside the `src/content/posts` directory:
+
+<FileTree title="Project Structure">
+- src/
+  - content/
+    - posts/
+      - **my-first-post.mdoc**
+</FileTree>
+
+That Markdoc file should look something like this:
+
+```markdown
+---
+title: My First Post
+---
+
+This is my very first post. I am **super** excited.
+```
+
+Niiiice ✨
+
+Time to display that post on the frontend now.
+
+## Rendering Keystatic content
+
+:::tip
+While Keystatic provides its own [Reader and Renderer APIs](https://keystatic.com/docs/reader-api) to pull data for your frontend, it also works great with Astro's [content collections](/en/guides/content-collections/)!
+:::
+
+### Displaying a collection list
+
+Since you instructed Keystatic to manage `posts` in `src/content/posts`, they'll be available as Astro's content collections as the `posts` collection slug:
+
+```tsx {4}
+---
+import { getCollection } from 'astro:content'
+
+const posts = await getCollection('posts')
+---
+
+<main>
+  <pre>{JSON.stringify(posts, null, 2)}</pre>
+</main>
+
+```
+
+### Displaying a single collection entry
+
+You can also get a single collection entry by its slug:
+
+```tsx {4}
+---
+import { getEntry } from 'astro:content'
+
+const post = await getEntry('posts', 'my-first-post')
+---
+
+<main>
+  <pre>{JSON.stringify(post, null, 2)}</pre>
+</main>
+```
+
+### Rendering document field content
+
+If you examine the output from our `post` entry above, you will notice that the `body` field — which represents the `content` of our post — is displaying raw Markdown.
+
+But you probably want to render HTML instead on your page.
+
+Well, good news: Astro provides a `<Content />` component that can do the heavy lifting of converting rich text data into properly formatted markup:
```

Comment:
``````suggestion
Use Astro's Content Collections API to [query and display your posts and collections](en/guides/content-collections/#querying-collections), just as you would in any Astro project.

### Displaying a collection list

The following example displays a list of each post title, with a link to an individual post page.

```tsx {4}
---
import { getCollection } from 'astro:content'

const posts = await getCollection('posts')
---
<ul>
  {posts.map(post => (
    <li>
      <a href={`/posts/${post.slug}`}>{post.data.title}</a>
    </li>
  ))}
</ul>
```

### Displaying a single entry

To display content from an individual post, you can import and use the `<Content />` component to [render your content to HTML)(/en/guides/content-collections/#rendering-content-to-html):
``````
OK... :sweat_smile: 

I really think at this point, we're back on Astro territory, and... I think we can get away with just saying, "Dude! It's content collections now!" So, see how this feels...

Reply:
I agree. One really neat thing about Keystatic is it can just do its thing and get out of the way. Letting Astro handle things the Astro way is the _normal_ pathway here, and treating it as the normal thing is indeed adequate 👍